### PR TITLE
Fix multi assignee parsing

### LINKED ISSUE #882: Parser accepts invalid code
## Description

The pest parser accepts code that contains array access in the declaration, which is invalid code.

## Environment

- Compiler version: 0.7.3
- Operating system: Ubuntu 20.04 LTS

## Steps to Reproduce

```zokrates
def main():
    field[2] a[2] = [1, 2]
    return
```

will result in

```
Compiling ./test.zok

assertion failed: `(left == right)`
  left: `1`,
  right: `0` (zokrates_core/src/absy/from_ast.rs:252:21)
```

### LINKED ISSUE #885: Tuple assignment returns wrong error message
```python
def swap(u32 a, u32 b, bool c) -> (u32, u32):
        a = if c then a else b fi
        b = if c then b else a fi
        return a, b

def main(u32[10] a) -> u32[10]:
        for u32 i in 0..10 do
                for u32 j in 0..9 do
                        bool need_swap = a[j + 1] < a[j]
                        a[j], a[j + 1] = swap(a[j], a[j + 1], need_swap)
                endfor
        endfor
        return a
```

This results in the following error:
```
	Function definition for function swap with signature (u32, u32, bool) -> (u32[10], u32[10]) not found.
```

I think it infers the wrong function signature, because `a[j]` is u32 and not u32[10]. Maybe this expression is not allowed in the first place, but it definitely comes up with the wrong expectation.

### PR DESCRIPTION
**Pull request by Schaeff**
fixes #882 and fixes #885

### CODE REVIEW

File: zokrates_core/src/compile.rs [hunk 1]
```diff
@@ -237,6 +237,8 @@ fn check_with_arena<'ast, T: Field, E: Into<imports::Error>>(
     let typed_ast = Checker::check(compiled)
         .map_err(|errors| CompileErrors(errors.into_iter().map(CompileError::from).collect()))?;
 
+    println!("{}", typed_ast);
```
Comment (dark64) [inline zokrates_core/src/compile.rs]: Remove this leftover debug code

File: zokrates_cli/examples/assign_to_element.zok [hunk 2]
```diff
@@ -0,0 +1,11 @@
+def foo() -> u32:
+        return 0
```
Comment (dark64) [inline zokrates_cli/examples/assign_to_element.zok]: Not really necessary, but we should stick to "tabs" being 4 spaces for the sake of cleaner formatting and consistency

**Review by dark64 — COMMENTED** [review]
Some minor tweaks, but otherwise seems good to me!

File: zokrates_cli/examples/assign_to_element.zok (hunk 2 above)
Comment (Schaeff) [reply zokrates_cli/examples/assign_to_element.zok]: Can we enforce this somehow? I don't care what we use as long as it's consistent and CI enforced

Comment (Schaeff) [reply zokrates_cli/examples/assign_to_element.zok]: Opened #901 for this and fixed this manually here

**Comment by Schaeff** [comment]
@dark64 can you have another glance at this one?

**Review by dark64 — APPROVED** [review]
:100:

**Pull request merged**